models/model-utils.js

```diff
@@ -0,0 +1,13 @@
+const db = require("../db/connection");
+
+exports.checkForeignPrimaryKey = (foreignTable, columnName, id) => {
+  // table and columnName are defined within the server, not by the user,
+  // so SQL Injection pretection not required - A hacker with sufficient
+  // access to pass in a malicious query would mean they have access to the
+  // database without needing this function
+
+  const query = `SELECT * FROM ${foreignTable} WHERE ${columnName} = $1`;
```

Comment:
You _could_ use a greenList to verify that that the foreignTable and columnName are valid?

Reply:
For me that's unnecessary code as noted in the comments - I could do but I think I'll leave it. I hope that's ok!